libs/metamask/src/lib/+state/erc1155.ts

```diff
@@ -11,59 +11,53 @@ interface ERC1155_Event {
   URI: (value: string, id: number) => void;
 }
 
+export interface ERC1155_Meta {
+  name: string;
+  description: string;
+  image: string;
+  image_data?: string;
+  background_color?: string;
+  animation_url?: string;
+  youtube_url?: string;
+  attributes: { display_type?: string, trait_type: string, value: string }[];
+}
+
+export interface ERC1155_Token {
+  id: string;
+  account: string;
+  balance: number;
+  meta: ERC1155_Meta;
+}
 
 @Injectable({ providedIn: 'root' })
 export class ERC1155 extends Contract {
-
   constructor(metamask: MetamaskService) {
     if (!metamask.signer) throw new Error('ERC1155 requires a signer');
     super(env.eth.erc1155, abi, metamask.signer);
   }
 
-  on<K extends keyof ERC1155_Event>(event: K, listener: ERC1155_Event[K]) {
-    return super.on(event, listener)
-  }
-  off<K extends keyof ERC1155_Event>(event: K, listener: ERC1155_Event[K]) {
-    return super.off(event, listener)
-  }
-  once<K extends keyof ERC1155_Event>(event: K, listener: ERC1155_Event[K]) {
-    return super.once(event, listener)
-  }
-  removeAllListeners(event?: Extract<keyof ERC1155_Event, string> | EventFilter) {
-    return super.removeAllListeners(event);
-  }
-  
-  /**
-    Transfers `value` amount of an `id` from the `from` address to the `to` address specified (with safety call).
-    @param from    Source address
-    @param to      Target address
-    @param id      ID of the token type
-    @param value   Transfer amount
-    @param data    Additional data with no specified format, MUST be sent unaltered in call to `onERC1155Received` on `to`
-  */
-  safeTransferFrom(from: string, to: string, id: number, value: number, data: string): Promise<Transaction> {
-    return super.safeBatchTransferFrom(from, to, id, value, data);
+  async getTokens(account: string) {
+    const ids = await this.getTokenIds(account);
+    const promises = ids.map(async (id: number) => {
+      const [ balance, meta ] = await Promise.all([
+        this.balanceOf(account, id),
+        this.getMeta(id)
+      ]);
+      return { id, account, balance, meta };
+    });
+    return Promise.all(promises);
   }
 
-  /**
-    Transfers `values` amount(s) of `ids` from the `from` address to the `to` address specified (with safety call).
-    @param from    Source address
-    @param to      Target address
-    @param ids     IDs of each token type (order and length must match values array)
-    @param values  Transfer amounts per token type (order and length must match ids array)
-    @param data    Additional data with no specified format, MUST be sent unaltered in call to the `ERC1155TokenReceiver` hook(s) on `to`
-  */
-  safeBatchTransferFrom(from: string, to: string, ids: number[], values: number[], data: string): Promise<Transaction> {
-    return super.safeBatchTransferFrom(from, to, ids, values, data);
+  getMeta(id: number): Promise<ERC1155_Meta> {
+    return this.uri(id)
```

Comment:
any reason to use then instead of async/await ?

Reply:
no reason, I find it cleaner like that. But async/await works the same